### PR TITLE
Make PerGameOptions be a different class for each game

### LINKED ISSUE #2844: Options: support per-game custom fields
Currently it assumes all games have the same fields

### PR DESCRIPTION
**Pull request by henriquegemignani**
Fixes #2844.

### CODE REVIEW

File: randovania/games/super_metroid/gui/dialog/super_cosmetic_patches_dialog.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (duncathan) [inline randovania/games/super_metroid/gui/dialog/super_cosmetic_patches_dialog.py:1]: ?

File: test/games/dread/gui/test_dread_game_export_dialog.py [hunk 2]
```diff
@@ -127,9 +138,11 @@ def test_on_input_file_button(skip_qtbot, tmp_path, mocker):
 def test_get_game_export_params(skip_qtbot, tmp_path):
     # Setup
     options = MagicMock()
-    options.options_for_game.return_value.input_path = tmp_path.joinpath("input/game.iso")
-    options.options_for_game.return_value.output_directory = tmp_path.joinpath("output")
-    options.options_for_game.return_value.output_format = "iso"
+    options.options_for_game.return_value = DreadPerGameOptions(
+        cosmetic_patches=DreadCosmeticPatches.default(),
+        input_directory=tmp_path.joinpath("input/game.iso"),
```
Comment (duncathan) [inline test/games/dread/gui/test_dread_game_export_dialog.py:143]: shouldn't this be a directory, not a file?

File: randovania/games/super_metroid/gui/dialog/super_cosmetic_patches_dialog.py (hunk 1 above)
Comment (henriquegemignani) [reply randovania/games/super_metroid/gui/dialog/super_cosmetic_patches_dialog.py:1]: Probably an accidental enter

File: test/games/dread/gui/test_dread_game_export_dialog.py (hunk 2 above)
Comment (henriquegemignani) [reply test/games/dread/gui/test_dread_game_export_dialog.py:143]: Yes, but `get_game_export_params` doesn't really care if the input is valid.

**Review by duncathan — APPROVED** [review]

**Pull request merged**